File: content/kotlin/current/source/includes/connect/ServerSelection.kt
import com.mongodb.ConnectionString
import com.mongodb.MongoClientSettings
import com.mongodb.connection.ClusterDescription
import com.mongodb.connection.ServerDescription
import com.mongodb.connection.ServerType
import com.mongodb.selector.ServerSelector
import com.mongodb.kotlin.client.coroutine.MongoClient

// start-custom-selector
class CustomServerSelector : ServerSelector {
    override fun select(cluster: ClusterDescription): List<ServerDescription> {
        return cluster.serverDescriptions.filter { it.type == ServerType.REPLICA_SET_SECONDARY }
    }
}
// end-custom-selector

fun main() {
    // start-selector
    val settings = MongoClientSettings.builder()
        .applyConnectionString(ConnectionString("<connection URI>"))
        .applyToClusterSettings { builder ->
            builder.serverSelector(CustomServerSelector())
        }
        .build()

    val client = MongoClient.create(settings)
    // end-selector
}

File: content/csharp/v3.x/source/includes/fundamentals/code-examples/connection/ServerSelection.cs
using MongoDB.Driver;
using MongoDB.Driver.Core.Clusters;
using MongoDB.Driver.Core.Clusters.ServerSelectors;
using MongoDB.Driver.Core.Servers;

public class ServerSelection
{
    // Replace with your connection string
    private const string MongoConnectionString = "<connection URI>";

    public static void Main(string[] args)
    {
        {
            // start-server-selector
            var settings = MongoClientSettings.FromConnectionString("<connection string>");
            var clusterConfigurator = builder =>
            {
                builder.ConfigureCluster(c =>
                    c.With(PreServerSelector: new RandomServerSelector()));
            };
            
            settings.ClusterConfigurator = clusterConfigurator;
            var client = new MongoClient(settings);
            // end-server-selector
        }
    }
}

// start-custom-class
public class CustomServerSelector : IServerSelector
{
    public IEnumerable<ServerDescription> SelectServers(ClusterDescription cluster,
        IEnumerable<ServerDescription> servers)
    {
        return servers.Where(server => server.Type == ServerType.ReplicaSetSecondary);
    }
}
// end-custom-class
